worker/src/ocall_bridge/remote_attestation_ocall.rs

```diff
@@ -72,87 +67,40 @@ impl RemoteAttestationBridge for RemoteAttestationOCall {
         spid: sgx_spid_t,
         quote_nonce: sgx_quote_nonce_t,
     ) -> OCallBridgeResult<(sgx_report_t, Vec<u8>)> {
-        let mut real_quote_len: u32 = 0;
-
-        let (p_sig_rl, sig_rl_size) = vec_to_c_pointer_with_len(revocation_list);
-
-        let ret =
-            unsafe { sgx_calc_quote_size(p_sig_rl, sig_rl_size, &mut real_quote_len as *mut u32) };
-
-        ensure!(
-            ret == sgx_status_t::SGX_SUCCESS,
-            OCallBridgeError::GetQuote(ret)
-        );
-
-        debug!("    Quote size = {}", real_quote_len);
-
-        let p_report = &report as *const sgx_report_t;
-        let p_spid = &spid as *const sgx_spid_t;
-        let p_nonce = &quote_nonce as *const sgx_quote_nonce_t;
-
-        let mut qe_report = sgx_report_t::default();
-        let p_qe_report = &mut qe_report as *mut sgx_report_t;
-
-        const RET_QUOTE_BUF_LEN: usize = 2048;
-        let mut return_quote_buf = [0u8; RET_QUOTE_BUF_LEN];
-        let p_quote = return_quote_buf.as_mut_ptr();
-
-        if real_quote_len > RET_QUOTE_BUF_LEN as u32 {
-            error!(
-                "   effective quote length ({}) exceeds buffer size ({})",
-                real_quote_len, RET_QUOTE_BUF_LEN
-            );
-            return Err(OCallBridgeError::GetQuote(
-                sgx_status_t::SGX_ERROR_FAAS_BUFFER_TOO_SHORT,
-            ));
-        }
-
-        let ret = unsafe {
-            sgx_get_quote(
-                p_report,
+        let real_quote_len = self
+            .enclave_api
+            .calc_quote_size(revocation_list.clone())
+            .map_err(|e| match e {
+                substratee_enclave_api::error::Error::Sgx(s) => OCallBridgeError::GetQuote(s),
+                _ => OCallBridgeError::GetQuote(sgx_status_t::SGX_ERROR_UNEXPECTED),
+            })?;
+
+        self.enclave_api
+            .get_quote(
+                revocation_list,
+                report,
                 quote_type,
-                p_spid,
-                p_nonce,
-                p_sig_rl,
-                sig_rl_size,
-                p_qe_report,
-                p_quote as *mut sgx_quote_t,
+                spid,
+                quote_nonce,
                 real_quote_len,
             )
```

Comment:
We can now use the enclave-api here to make the necessary e-calls for SGX remote attestation